CodeEdit/Features/Settings/Pages/ThemeSettings/Models/ThemeModel+CRUD.swift

```diff
@@ -0,0 +1,292 @@
+                            self.selectedTheme = self.selectedDarkTheme
+                        } else {
+                            self.selectedTheme = self.selectedLightTheme
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    func importTheme() {
+        let openPanel = NSOpenPanel()
+        let allowedTypes = [UTType(filenameExtension: "cetheme")!]
+
+        openPanel.prompt = "Import"
+        openPanel.allowedContentTypes = allowedTypes
+        openPanel.canChooseFiles = true
+        openPanel.canChooseDirectories = false
+        openPanel.allowsMultipleSelection = false
+
+        openPanel.begin { result in
+            if result.rawValue == NSApplication.ModalResponse.OK.rawValue {
+                if let url = openPanel.urls.first {
+                    self.duplicate(url)
+                }
+            }
+        }
+    }
+
+    func duplicate(_ url: URL) {
+        do {
+            self.isAdding = true
+            // Construct the destination file URL
+            var destinationFileURL = self.themesURL.appendingPathComponent(url.lastPathComponent)
+
+            // Extract the base filename and extension
+            let fileExtension = destinationFileURL.pathExtension
+
+            var fileName = destinationFileURL.deletingPathExtension().lastPathComponent
+            var newFileName = fileName
+
+            // Check if the file already exists
+            var iterator = 1
+
+            let isBundled = url.absoluteString.hasPrefix(bundledThemesURL?.absoluteString ?? "")
+            let isImporting =
+                !url.absoluteString.hasPrefix(bundledThemesURL?.absoluteString ?? "")
+                && !url.absoluteString.hasPrefix(themesURL.absoluteString)
+
+            if isBundled {
+                newFileName = "\(fileName) \(iterator)"
+                destinationFileURL = self.themesURL
+                    .appendingPathComponent(newFileName)
+                    .appendingPathExtension(fileExtension)
+            }
+
+            while FileManager.default.fileExists(atPath: destinationFileURL.path) {
+                fileName = destinationFileURL.deletingPathExtension().lastPathComponent
+
+                // Remove any existing iterator
+                if let range = fileName.range(of: " \\d+$", options: .regularExpression) {
+                    fileName = String(fileName[..<range.lowerBound])
+                }
+
+                // Generate a new filename with an iterator
+                newFileName = "\(fileName) \(iterator)"
+                destinationFileURL = self.themesURL
+                    .appendingPathComponent(newFileName)
+                    .appendingPathExtension(fileExtension)
+
+                iterator += 1
+            }
+
+            // Copy the file from selected URL to the destination
+            try FileManager.default.copyItem(at: url, to: destinationFileURL)
+
+            try self.loadThemes()
```

Comment:
Why are we loading all the themes into memory?

Reply:
So we have the data to list them all in settings. Is there an alternate way of handling this?

Reply:
My concern is that users aren't on the settings page most of the time, so having the themes in memory will be unused the majority of the application's lifetime. Instead we should load in the themes only when necessary.

For how VSCode does it, in `themes.contribution.ts`, the `InstalledThemesPicker` class has a function called `openQuickPick` where it creates an instance of the `MarketplaceThemesPicker` class that then loads in the themes that the user installed.

Reply:
I'm 100% on board with changing how this works, but that goes outside of the scope of this PR which aims to provide users with features to manage themes. Let's create a separate issue explaining specifically how this might work.

Reply:
Agreed.